llvm/lib/Transforms/Scalar/LowerMatrixIntrinsics.cpp

```diff
@@ -2198,6 +2210,49 @@ class LowerMatrixIntrinsics {
     return true;
   }
 
+  /// Lower uniform shape intrinsics, if shape information is available.
+  bool VisitUniformIntrinsic(IntrinsicInst *Inst) {
+    auto I = ShapeMap.find(Inst);
+    if (I == ShapeMap.end())
```

Comment:
I think this check shouldn't be needed, we should only be calling this for instructions with shape-info. I put up https://github.com/llvm/llvm-project/pull/142416 to clean this up in the existing functions